File: src/main/java/net/matees/settings/SettingsMenu.java
package net.matees.settings;

import org.bukkit.event.inventory.InventoryClickEvent;

import me.kodysimpson.simpapi.exceptions.MenuManagerException;
import me.kodysimpson.simpapi.exceptions.MenuManagerNotSetupException;
import me.kodysimpson.simpapi.menu.Menu;
import me.kodysimpson.simpapi.menu.PlayerMenuUtility;
import net.matees.Minigame;
import net.matees.arcade.itemrush.ItemRush;

public abstract class SettingsMenu extends Menu {

    public SettingsMenu(PlayerMenuUtility playerMenuUtility) {
        super(playerMenuUtility);
    }

    public abstract Minigame getMinigame();

    @Override
    public void handleMenu(InventoryClickEvent inventoryClickEvent)
            throws MenuManagerNotSetupException, MenuManagerException {
        for (Setting setting : this.getMinigame().getSettings()) {
            if (setting.getMenuItem().isSimilar(inventoryClickEvent.getCurrentItem())) {
                setting.handleItemClick(inventoryClickEvent);

                // Updates?
                inventory.clear();
                setting.setMenuItem(setting.getMenuItem());
                this.setMenuItems();
            }
        }
    }

    @Override
    public int getSlots() {
        return 27;
    }

    @Override
    public boolean cancelAllClicks() {
        return true;
    }

    @Override
    public void setMenuItems() {
        for (Setting setting : this.getMinigame().getSettings()) {
            inventory.setItem(setting.getMenuItemSlot(), setting.getMenuItem());
        }
    }
}

File: src/main/java/net/matees/settings/SettingsMenu.kt
package net.matees.settings

import me.kodysimpson.simpapi.exceptions.MenuManagerException
import me.kodysimpson.simpapi.exceptions.MenuManagerNotSetupException
import me.kodysimpson.simpapi.menu.Menu
import me.kodysimpson.simpapi.menu.PlayerMenuUtility
import net.matees.arcade.Minigame
import org.bukkit.event.inventory.InventoryClickEvent

abstract class SettingsMenu(playerMenuUtility: PlayerMenuUtility?) : Menu(playerMenuUtility) {
    abstract val minigame: Minigame

    @Throws(MenuManagerNotSetupException::class, MenuManagerException::class)
    override fun handleMenu(inventoryClickEvent: InventoryClickEvent) {
        for (setting in minigame.settings) {
            if (setting.menuItem.isSimilar(inventoryClickEvent.currentItem)) {
                setting.handleItemClick(inventoryClickEvent)

                // Updates?
                inventory.clear()
                setting.menuItem = setting.menuItem
                this.setMenuItems()
            }
        }
    }

    override fun getSlots(): Int {
        return 27
    }

    override fun cancelAllClicks(): Boolean {
        return true
    }

    override fun setMenuItems() {
        for (setting in minigame.settings) {
            inventory.setItem(setting.menuItemSlot, setting.menuItem)
        }
    }
}
